scripts/install-binaries.sh

```diff
@@ -0,0 +1,79 @@
+#!/bin/bash
+# Copyright (c) 2024 Red Hat, Inc.
+# Copyright Contributors to the Open Cluster Management project
+
+# The functions in this script are used to install the various binaries that
+# are required for the CI/CD pipeline to function, as well as execution of local e2e tests.
+# If the binaries are already executable on the ${PATH} of the host, the script will skip the installation.
+# Each function takes a path as the first argument, which is the directory where the binary will be installed.
+# If no path is provided, fallback to ${BIN_DIR} or default path to /usr/local/bin.
+
+OPERATOR_SDK_VERSION="${KUBECTL_VERSION:=v1.4.2}"
+KUBECTL_VERSION="${KUBECTL_VERSION:=v1.28.2}"
+KUSTOMIZE_VERSION="${KUSTOMIZE_VERSION:=v5.3.0}"
+JQ_VERSION="${JQ_VERSION:=1.6}"
+
+BIN_DIR="${BIN_DIR:=/usr/local/bin}"
+
+install_operator_sdk() {
+  bin_dir=${1:-${BIN_DIR}}
+  if ! command -v operator-sdk &>/dev/null; then
+    echo "operator-sdk not found on path, installing operator-sdk version ${OPERATOR_SDK_VERSION}"
+    if [[ "$(uname)" == "Linux" ]]; then
+      curl -L "https://github.com/operator-framework/operator-sdk/releases/download/${OPERATOR_SDK_VERSION}/operator-sdk_linux_amd64" -o operator-sdk
+    elif [[ "$(uname)" == "Darwin" ]]; then
+      curl -L "https://github.com/operator-framework/operator-sdk/releases/download/${OPERATOR_SDK_VERSION}/operator-sdk_darwin_$(uname -m)" -o operator-sdk
+    fi
```

Comment:
Yes, these are mostly intended to be used in CI. I expect most will have their own installs locally. I'm trying to consolidate with what exists in the repo already with as few changes as possible but will revisit everything when I've removed all the duplicate code.

Reply:
Indeed, but if you read the .MD in this PR it will become clear why that is not an option at the moment. Need to figure out what can break if we upgrade to later versions. As of now, from our understanding there is non-compliance with OpenShift 3.x clusters!

Reply:
I also agree with your point about removing the if statement. This is just a lift and shift and tidy for some code which is duplicated in multiple places across the repo. I am trying to minimise the changes in each PR as I clean up so bare with me :)

Reply:
Also mostly expect that users will already have these tools locally and its mostly just for CI right now.